test/proxy/transparent/ProxyAdmin.test.js

```diff
@@ -19,12 +20,12 @@ contract('ProxyAdmin', function (accounts) {
 
   beforeEach(async function () {
     const initializeData = Buffer.from('');
-    this.proxyAdmin = await ProxyAdmin.new(proxyAdminOwner);
-    const proxy = await TransparentUpgradeableProxy.new(
-      this.implementationV1.address,
-      this.proxyAdmin.address,
-      initializeData,
-    );
+    const proxy = await TransparentUpgradeableProxy.new(this.implementationV1.address, proxyAdminOwner, initializeData);
+
+    const proxyNonce = await web3.eth.getTransactionCount(proxy.address);
+    const proxyAdminAddress = computeCreateAddress(proxy.address, proxyNonce - 1); // Nonce already used
```

Comment:
you could get the proxyAdminAddress by reading the ERC1967 slot

Reply:
I prefer this way because we are testing the assumption that the ProxyAdmin is deployed by the proxy.

Reply:
We need to test both things:
- that the ProxyAdmin is deployed by the proxy
- that the ERC1967 slot is set correctly

So we'll have to get both values and compare them. I would have read the slot here, and add an `it` that check it matched the create prediction ... but doing the opposite also works.

Reply:
We're already testing both cases in the following test:

```js
it('sets the proxy admin in storage with the correct initial owner', async function () {
  expect(await getAddressInSlot(this.proxy, AdminSlot)).to.be.equal(this.proxyAdminAddress);
  const proxyAdmin = await Ownable.at(this.proxyAdminAddress);
  expect(await proxyAdmin.owner()).to.be.equal(initialOwner);
});
```